### PR TITLE
#70 - email configuration

### PR DESCRIPTION
**Pull request by tavoda**
Passbolt chart changes:
- Email configuration with env variables in deployment
- Upgrade ingress to new kubernetes api version
- Readme update for passbolt

### CODE REVIEW

File: charts/passbolt/templates/deployment.yaml [hunk 1]
```diff
@@ -56,6 +56,22 @@ spec:
             - name: DATASOURCES_DEFAULT_PASSWORD
               value: {{ .Values.mariadb.auth.password }}
             {{- end }}
+            - name: EMAIL_TRANSPORT_DEFAULT_HOST
+              value: "{{ .Values.email.host }}"
+            - name: EMAIL_TRANSPORT_DEFAULT_PORT
+              value: {{ .Values.email.port }}
```
Comment (Thomah) [inline charts/passbolt/templates/deployment.yaml:62]: Be careful : you must always use a string for environment values. 

You may replace this section with :
```
- name: EMAIL_TRANSPORT_DEFAULT_PORT
  value: "{{ .Values.email.port }}"
```

File: charts/passbolt/templates/deployment.yaml [hunk 2]
```diff
@@ -56,6 +56,22 @@ spec:
...
+            {{- if .Values.email.username }}
+            - name: EMAIL_TRANSPORT_DEFAULT_USERNAME
+              value: "{{ .Values.email.username }}"
+            {{- end }}
+            {{- if .Values.email.password }}
+            - name: EMAIL_TRANSPORT_DEFAULT_PASSWORD
+              value: "{{ .Values.email.password }}"
+            {{- end }}
+            - name: EMAIL_TRANSPORT_DEFAULT_TIMEOUT
+              value: {{ .Values.email.timeout }}
```
Comment (Thomah) [inline charts/passbolt/templates/deployment.yaml:74]: Same as previous comment

File: charts/passbolt/templates/ingress.yaml [hunk 3]
```diff
@@ -26,16 +26,20 @@ spec:
       secretName: {{ .secretName }}
     {{- end }}
   {{- end }}
+  ingressClassName: {{ .Values.ingress.className }}
```
Comment (Thomah) [inline charts/passbolt/templates/ingress.yaml:29]: `ingressClassName` is not required if it's defined on the cluster level.

You should surround this line with a condition like
```
{{- if .Values.ingress.ingressClassName }}
```
in order to use the ingress class by default

File: charts/passbolt/templates/ingress.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 {{- $fullName := include "passbolt.fullname" . -}}
 {{- $svcPort := .Values.service.port -}}
 {{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
+apiVersion: networking.k8s.io/v1
```
Comment (Thomah) [inline charts/passbolt/templates/ingress.yaml:5]: In Kubernetes 1.14, there is no kind Ingress in this API version.

Please keep v1beta1 or add an other test on `.Capabilities.KubeVersion.GitVersion`. If you choose the last option, you also need to adapt the spec below.

Checkout this comment for details : https://github.com/kubernetes/kubernetes/issues/90077#issuecomment-718778960

**Review by Thomah — CHANGES_REQUESTED** [review]
Hi @tavoda 

Thanks for the PR ! There are just a few things that needs to be taken care of before merging. Please checkout the comments for details.

Also we need to increment `version` in `Chart.yaml` to make the release work. Can you update to `1.1.3` please ?

**Comment by tavoda** [comment]
Fixed in 2nd commit, hope everything fixed :crossed_fingers:

**Comment by Thomah** [comment]
Looks good to me 😃

Can you just increment version in Chart.yaml to 1.1.3 please ? The release will fail if not.

**Comment by tavoda** [comment]
My first helm chart PR, I'm not aware of this details and dependencies. Thanks for your help. Done

**Review by Thomah — APPROVED** [review]

**Comment by Thomah** [comment]
> My first helm chart PR, I'm not aware of this details and dependencies. Thanks for your help. Done

Don't mind it. Thanks for your contribution !

**Pull request merged**